writer/influxdb2/influxdb2.go

```
@@ -0,0 +1,59 @@
+package influxdb2
+
+import (
+	"context"
+	"time"
+
+	"github.com/aptible/mini-collector/batch"
+	"github.com/aptible/mini-collector/emitter/writer"
+	influxdb2 "github.com/influxdata/influxdb-client-go/v2"
+)
+
+const (
+	timeout = 30 * time.Second
+)
+
+type influxdbClient interface {
+	Write(org string, bucket string, batch batch.Batch) error
+	Close()
+}
+
+type influxdbWriter struct {
+	bucket string
+	org    string
+
+	client influxdbClient
+}
+
+type client struct {
+	influxdb2.Client
+}
+
+func (c *client) Write(org string, bucket string, batch batch.Batch) error {
+	bp := buildBatchPoints(batch.Entries)
+	return c.Client.WriteAPIBlocking(org, bucket).WritePoint(context.Background(), bp...)
+}
+
+func (c *client) Close() {
+	c.Client.Close()
+}
+
+func Open(config *Config) (writer.CloseWriter, error) {
+	options := influxdb2.DefaultOptions().SetMaxRetryTime(uint(timeout))
+	c := influxdb2.NewClientWithOptions(config.Address, config.AuthToken, options)
+	httpClient := &client{c}
+
+	return &influxdbWriter{
+		bucket: config.Bucket,
+		client: httpClient,
+	}, nil
+}
+
+func (w *influxdbWriter) Write(batch batch.Batch) error {
+	return w.client.Write(w.org, w.bucket, batch)
+}
+
+func (w *influxdbWriter) Close() error {
+	w.client.Close()
+	return nil
```

this is kinda weird, you're closing and always returning a nil. might wanna be careful about this if you want to actually return an error for any reason

---

`Close()` doesn't return anything, so I'm just doing this to conform to expected interface